### ISSUE TITLE
How do I update tsconfig.json

### ISSUE THREAD

**Issue by jesperronn** [mgechev/angular2-seed#219]
More of a question. How exactly do I keep my `tsconfig.json` file updated?

I tried running `npm run postinstall`, but I am not sure it catches all changes. 

(postinstall job runs `tsd reinstall --clean && tsd link && tsd rebundle` and more)

PS> I tried to add a .ts file somewhere, and saw that no line was added to the `tsconfig.json` file. But when I tried to update `package.json`, then the file updated.

What is supposed to update the tsdconfig.json, and how does that happen.

**Comment by mgechev** [comment]
Some IDEs and text editors does this automatically.
I have the same issue since I'm using vim. I'd suggest using paths with wildcards in the files array.

The problem comes when a new files array is being generated by the IDE/text editor of another contributor. In such case what we have here in the repo will be overwritten. How about, updating the tsconfig.json to something like:

```json
{
    "version": "1.6.0",
    "compilerOptions": {
        "target": "es5",
        "module": "commonjs",
        "declaration": false,
        "noImplicitAny": false,
        "removeComments": true,
        "noLib": false,
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "sourceMap": true
    },
    "files": [
        "app/**/*.ts",
        "gulpfile.ts",
        "tools/**/*.ts"
    ],
    "filesGlob": [
        "./app/**/*.ts",
        "./gulpfile.ts",
        "./tools/**/*.ts"
    ],
    "compileOnSave": false,
    "exclude": []
}
```

After we update it to the content above we can add it to `.gitignore` and prevent it from further changes.

**Comment by ludohenin** [comment]
Looks good.
What about this:
```
{
    "version": "1.6.0",
    "compilerOptions": {
        "target": "es5",
        "module": "commonjs",
        "declaration": false,
        "noImplicitAny": false,
        "removeComments": true,
        "noLib": false,
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "sourceMap": true
    },
    "files": [
        "**/*.ts"
    ],
    "filesGlob": [
        "./**/*.ts"
    ],
    "compileOnSave": false,
    "exclude": ["node_modules"]
}
```

**Comment by mgechev** [comment]
Yes, I like it

**Comment by jesperronn** [comment]
@mgechev @ludohenin Even though `files` and `filesGlob` are simpler in your suggestions, how can we get closer to this example:
https://github.com/juliemr/ng2-test-seed/blob/master/tsconfig.json

(Julie Ralphs example has none of those).

Excuse my lack of knowledge, but if that works, couldn't we just remove them?

**Comment by jesperronn** [comment]
oh [sidenote]: and `compileOnSave` does not exist in the configuration options. do you have a link to an explanation on how its used and by whom?

**Comment by ludohenin** [comment]
Here is some doc about tsconfig https://github.com/TypeStrong/atom-typescript/blob/master/docs/tsconfig.md
About `files` and `fileGlob` at some point it was required to have files listed in order to avoid having `/// <ref...` tags in ts files. Mybe this is not required anymore. Then Julie's setup is fine.

**Comment by ludohenin** [comment]
Addressed by #224

**Comment by jesperronn** [comment]
Thanks @ludohenin  :+1:
